### PR TITLE
BUG: Fix Li threshold calculator for negative image values

### PR DESCRIPTION
**Pull request by hjmjohnson**
The histogram values are shifted by the minimum bin value to prevent
taking a log of a negative number.

### CODE REVIEW

**Comment by hjmjohnson** [comment]
http://review.source.kitware.com/#/c/23612/

Bradley Lowekamp
Uploaded patch set 1.Aug 2 2:35 PM

Bradley Lowekamp
Uploaded patch set 2.Aug 2 3:12 PM

Kitware Build Robot
Patch Set 1: Verified-1 Build Failed: CDash filtered results: https://open.cdash.org/index.php?&project=Insight&filtercount=3&field1=buildname/string&compare1=63&value1=23612-1&field2=buildstarttime/date&compare2=83&value2=2015-03-01&field3=buildstarttime/date&compare3=84&value3=2029-01-01Aug 2 4:31 PM

Kitware Build Robot
Patch Set 2: Verified-1 Build Failed: CDash filtered results: https://open.cdash.org/index.php?&project=Insight&filtercount=3&field1=buildname/string&compare1=63&value1=23612-2&field2=buildstarttime/date&compare2=83&value2=2015-03-01&field3=buildstarttime/date&compare3=84&value3=2029-01-01Aug 2 4:43 PM

Matt McCormick
Aug 24 10:24 AM
↩
Patch Set 2: Code-Review-1

Todos:

Add test with negative image pixel intensities and crashes the class
Update baseline for Li test
Rebase on master to avoid unrelated issues

**Comment by blowekamp** [comment]
rebased the patch.

File: Modules/Filtering/Thresholding/include/itkLiThresholdCalculator.hxx [hunk 1]
```diff
@@ -114,6 +122,11 @@ LiThresholdCalculator<THistogram, TOutput>
   //
   //#define IS_NEG( x ) ( ( x ) < -DBL_EPSILON )
   //
+
+  // Shift the mean by the minimum to have the range start at zero,
+  // and void the log of a negative value.
```
Comment (dzenanz) [inline Modules/Filtering/Thresholding/include/itkLiThresholdCalculator.hxx]: void -> avoid
